packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -521,3 +525,29 @@ gmaps.LatLng _pixelToLatLng(gmaps.GMap map, int x, int y) {
 
   return projection.fromPointToLatLng!(point)!;
 }
+
+String _sanitizeHtml(String htmlString) {
+  TrustedTypePolicy? trustedTypePolicy;
+
+  try {
+    // Firefox and Safari don't support Trusted Types yet.
+    // See https://developer.mozilla.org/en-US/docs/Web/API/TrustedTypePolicyFactory#browser_compatibility
+    trustedTypePolicy =
+        WindowWithTrustedTypes(window).trustedTypesNullable?.createPolicy(
+              'google_maps_flutter_sanitize',
+              TrustedTypePolicyOptions(
+                createHTML: (String html, JSAny? arguments) {
+                  return sanitizeHtml(html);
+                }.toJS,
+              ),
+            );
+  } catch (_) {
+    // Fallback to not using the trusted types policy.
+  }
+
+  if (trustedTypePolicy == null) {
+    return sanitizeHtml(htmlString);
+  }
+
+  return trustedTypePolicy.createHTML(htmlString, null).toString();
```

Comment:
Please don't convert the result of calling `createHTML` to String, or we'll be using `unsafe_html` APIs anyway.

If trustedTypes are available, we need to have a `safeInnerHtml` setter on the HTML Object that accepts the `TrustedHTML` object as its contents. For other browsers, we just use the regular innerHtml setter.

See this example:

* https://github.com/flutter/packages/blob/91d11d696196880fe3cd4817c08362e59d7b5c56/packages/google_identity_services_web/lib/src/js_loader.dart#L49-L53

Reply:
Added a setter for innerHTML that only accepts `TrustedHTML`.